app/models/solid_queue/claimed_execution.rb

```diff
@@ -60,12 +54,9 @@ def discard_all_from_jobs(*)
   def perform
     result = execute
 
-    if result.success?
-      finished
-    else
-      failed_with(result.error)
-      raise result.error
-    end
+    result.just? ? finished : failed_with(result.reason)
+
+    result
```

Comment:
What's the advantage of using `Concurrent::Maybe` here vs. a regular struct?

Reply:
Warning: Rationalization alert.

I dislike Struct.  I would have reached for Data, but when I started this PR, SQ was on Ruby 3.3.1.

The Result struct was nothing more than a simple Maybe.  It was also only used in 1 place, for 1 `if` and made me question if it was really needed at all (return Array[success, exception]).  But with the change from Futures to Promises::Futures, I also pushed the error reporting out of the execute method back to the Future. With some of the logic moving back to the Future, why keep the logic all within the Concurrent library?

If all of this is bothersome, you're not buying the rationalization for the change, of you simply feel that this doesn't make SQ better I'm happy to refactor the Maybe out and, now that SQ is using Ruby 3.3.5, bring the Result object back, but as a Data instead of Struct.